File: leetcode/arrays/leetcode2028.java
package arrays;

import java.util.*;

public class leetcode2028 {
    public static void main(String[] args) {
        leetcode2028 sol = new leetcode2028();
        System.out.println(Arrays.toString(sol.missingRolls(new int[]{3,2,4,3}, 4, 2)));
    }
    public int[] missingRolls(int[] rolls, int mean, int n) {
        int sum = 0 ;
        for(int roll:rolls){
            sum+=roll;
        }
        int remainingSum = mean*(n+rolls.length)-sum;
        if(remainingSum>6*n || remainingSum<n){
            return new int[]{};
        }
        int[] ans = new int[n];
        int distributedMean = remainingSum/n;
        int mod = remainingSum%n;
        Arrays.fill(ans,distributedMean);
        for(int i =0;i<mod;i++){
            ans[i]++;
        }
        return ans;
    }
}

File: leetcode/arrays/leetcode2028.py

class Solution(object):
    def missingRolls(self, rolls, mean, n):
        """
        :type rolls: List[int]
        :type mean: int
        :type n: int
        :rtype: List[int]
        """
        sum  = 0
        for roll in rolls:
            sum+=roll
        
        remaining_sum = mean*(n+len(rolls))-sum
        
        require_sum =(int) (remaining_sum/n)
        
        mod = remaining_sum%n
        
        ans = []
        for i in range(0,mod):
            ans.append(require_sum+1)
        for i in range(mod,n):
            ans.append(require_sum)
        
        return ans
    
sol = Solution()
print(sol.missingRolls([3,2,4,3],4,2))